compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1378,6 +1374,16 @@ private BLangRecordTypeNode getFrameTypeNode() {
         return frameTypeNode;
     }
 
+    private BLangExpression unwrapCheckedExpression(BLangExpression collectionExp) {
+        if (collectionExp.getKind() == NodeKind.GROUP_EXPR &&
```

Comment:
Dont we need unwraping for group expr separately, because this can fail if someone intentionally put double paranthesis right? e.i.. `(( check <some expr>))`